packages/trader/src/Stores/Modules/Trading/trade-store.js

```diff
@@ -1214,7 +1214,7 @@ export default class TradeStore extends BaseStore {
 
             // Sometimes the initial barrier doesn't match with current barrier choices received from API.
             // When this happens we want to populate the list of barrier choices to choose from since the value cannot be specified manually
-            if (this.is_vanilla) {
+            if (this.is_vanilla && contract_type === this.vanilla_trade_type) {
```

Comment:
What happens when there's max_stake or min_stake error from VANILLALONGPUT?

Reply:
It will work the same way as before. If the current contract_type is `vanillalongput`, it will only check if the error comes from `vanillalongput` call and vice versa. Previously, it didn't check for the contract type, so it would cause an infinite loop of values changed on each tick from the wrong contract_type.

Reply:
I was wondering because "vanilla_trade_type" only gets used in line 149 and 1197 and don't see it changed in trade-store

Reply:
i guess its changed elsewhere

Reply:
Yes, correct, it comes from UI store and changed in `VanillaTradeTypes` component. Basically, this value gets changed whenever the Call/Put tab is switched in the trade parameters sidebar.